device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceIO.java

```diff
@@ -218,6 +215,11 @@ public void close() throws IOException
     {
         synchronized (this.stateLock)
         {
+            if (state == IotHubConnectionStatus.DISCONNECTED)
```

Comment:
does `DISCONNECTED` status means all the tasks schedulers are shut down?

Reply:
Yes